### PR TITLE
Create basic flask service for creating movies from images

### PR DESCRIPTION
**Pull request by bnmajor**
Very rough first pass at flask service, this will need further iterations to complete. Currently still using some hard-coded data for ease of testing.

### CODE REVIEW

**Comment by bnmajor** [comment]
The movie is now created successfully and can be sent back to the client. The movie is currently set to 1 fps but this can be changed if/as needed.

File: flask/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Flask
+ffmpeg-python
+Flask-Cors
```
Comment (cjh1) [inline flask/requirements.txt:3]: Look like we need the girder client as well ...

File: flask/src/server.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+    gc = GirderClient(apiUrl='https://data.kitware.com/api/v1/')
+    gc.setToken(token)
+    output_file = tempfile.NamedTemporaryFile(suffix='.mp4', delete=True)
+    with tempfile.TemporaryDirectory() as tmpdir:
+        item = gc.getItem(id)
+        if not item:
+            return Response('Unable to fetch item', status=400, mimetype='application/json')
+        gc.downloadItem(id, tmpdir, item['name'])
+        path = tmpdir + '/' + item['name'] + '/*.svg'
```
Comment (cjh1) [inline flask/src/server.py:26]: Let us `pathlib` to build up this path

File: flask/src/server.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+app = Flask(__name__)
+
+CORS(app, resources={r'/*': {'origins': '*'}})
+
+@app.route('/api/movie/<id>', methods=['GET'])
+def create_movie(id):
+    token = request.headers.get('girderToken', '')
+    if not token or not id:
+        return Response('Invalid token or parameter ID.', status=400, mimetype='application/json')
```
Comment (cjh1) [inline flask/src/server.py:16]: What does this return? Just a string?

File: client/src/App.vue [hunk 4]
```diff
@@ -369,6 +370,25 @@ export default {
       this.numReady += 1;
       this.getRangeData(event);
     },
+
+    fetchMovie(id) {
+      axios({
+        url: `http://localhost:5000/api/movie/${id}`,
```
Comment (cjh1) [inline client/src/App.vue:376]: We should have some way to configure this at build time. No sure if vue has support for this?

File: flask/requirements.txt (hunk 1 above)
Comment (bnmajor) [reply flask/requirements.txt:3]: This has been added in

File: flask/src/server.py (hunk 2 above)
Comment (bnmajor) [reply flask/src/server.py]: Fixed

File: flask/src/server.py [hunk 5]
```diff
@@ -3,39 +3,37 @@
...
 
 from girder_client import GirderClient
-from flask import Flask, send_file, request, Response
+from flask import Flask, request, Response, send_file
 from flask_cors import CORS
 
 app = Flask(__name__)
 
-CORS(app, expose_headers=["x-suggested-filename"])
+cors_route = os.environ.get('CORS_API', '/*')
```
Comment (cjh1) [inline flask/src/server.py:11]: Its not the route we want to configure its the origin, where the request is coming from?

**Comment by bnmajor** [comment]
Updated the env variable to `cors_domain`

File: flask/src/server.py [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+import tempfile
+import os
+
+from girder_client import GirderClient
+from flask import Flask, request, Response, send_file
+from flask_cors import CORS
+
+app = Flask(__name__)
+
+cors_domain = os.environ.get('CORS_API', '*')
```
Comment (cjh1) [inline flask/src/server.py:11]: Sorry to be a pain but the env var should probably be CORS_DOMAIN as well.

Comment (bnmajor) [reply flask/src/server.py]: That makes sense, sorry for overlooking that. Fixed.

File: flask/src/server.py [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+from girder_client import GirderClient
+from flask import Flask, request, Response, send_file
+from flask_cors import CORS
+
+app = Flask(__name__)
+
+cors_domain = os.environ.get('CORS_DOMAIN', '*')
+girder_url = os.environ.get('GIRDER_API_URL', 'https://data.kitware.com/api/v1/')
+
+CORS(app, resources={r'/*': {'origins': cors_route}})
```
Comment (cjh1) [inline flask/src/server.py:14]: I get the following when I test this:

```
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 783, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/app/server.py", line 14, in <module>
    CORS(app, resources={r'/*': {'origins': cors_route}})
NameError: name 'cors_route' is not defined
```

```cors_route``` needs to be ```cors_domain```

Comment (bnmajor) [reply flask/src/server.py]: This has been fixed, testing was successful.

**Pull request merged**